code/modules/mapping/mapping_helpers.dm

```diff
@@ -380,3 +380,82 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 	log_mapping("[src] at [x],[y] could not find an airlock on current turf, cannot place paper note.")
 	qdel(src)
 
+/obj/effect/mapping_helpers/simple_pipes
+	name = "Simple Pipes"
+	late = TRUE
+	icon_state = "pipe-3"
+	var/piping_layer = 3
+	var/pipe_color = ""
+	var/connection_num = 0
+	var/hide = FALSE
+
+/obj/effect/mapping_helpers/simple_pipes/LateInitialize()
+	var/list/connections = list( dir2text(NORTH)  = FALSE, dir2text(SOUTH) = FALSE , dir2text(EAST) = FALSE , dir2text(WEST) = FALSE)
+	var/list/valid_connectors = typecacheof(/obj/machinery/atmospherics)
+	for(var/direction in connections)
+		var/turf/T = get_step(src,  text2dir(direction))
+		for(var/_type in T.contents)
+			if(istype(_type,type))
+				var/obj/effect/mapping_helpers/simple_pipes/found = _type
+				if(found.piping_layer != piping_layer)
+					continue
+				connections[direction] = TRUE
+				connection_num++
+				break
+			if(!is_type_in_typecache(_type,valid_connectors))
+				continue
+			var/obj/machinery/atmospherics/machine = _type
+
+			if(machine.piping_layer != piping_layer)
+				continue
+
+			if(angle2dir(dir2angle(text2dir(direction))+180) & machine.initialize_directions)
+				connections[direction] = TRUE
+				connection_num++
+				break
+
+	var/turf/this_turf = get_turf(src)
+	switch(connection_num)
+		if(1)
+			for(var/direction in connections)
+				if(connections[direction] != TRUE)
+					continue
+				spawn_pipe(direction)
+		if(2)
+			for(var/direction in connections)
+				if(connections[direction] != TRUE)
+					continue
+				//Detects straight pipes connected from east to west , north to south etc.
+				if(connections[dir2text(angle2dir(dir2angle(text2dir(direction))+180))] == TRUE)
+					spawn_pipe(direction)
+					break
+
+				for(var/direction2 in connections - direction)
+					if(connections[direction2] != TRUE)
+						continue
+					spawn_pipe(dir2text(text2dir(direction)+text2dir(direction2)))
+		if(3)
+			for(var/direction in connections)
+				if(connections[direction] == FALSE)
+					var/obj/machinery/atmospherics/pipe/manifold/pipe = new(this_turf,TRUE,text2dir(direction))
+					pipe.hide = hide
+					pipe.piping_layer = piping_layer
+					pipe.update_layer()
+					pipe.paint(pipe_color)
+		if(4)
+			var/obj/machinery/atmospherics/pipe/manifold4w/pipe = new(this_turf)
+			pipe.hide = hide
+			pipe.piping_layer = piping_layer
+			pipe.update_layer()
+			pipe.paint(pipe_color)
```

Comment:
This is annoying but I can't think of a better way

Reply:
This is just a helper, i bet noone will look into this file for the next 5 years when the better solution IS found, and they will be left wondering 'WTF were they thinking?!'